lib/pipeline.js

```diff
@@ -1276,19 +1276,19 @@ class PipelineModel extends BaseModel {
 
         if (!config.aggregate) {
             const events = await this.getEvents(options);
-            const metrics = [];
-            const promiseArray = events.map(async (e) => {
+            const metrics = await Promise.all(events.map(async (e) => {
```

Comment:
what's the purpose of `Promise.all` here `events.map` already resolve to a series of event metrics?

Reply:
cannot use `await` inside regular `events.map` because it's not async: https://stackoverflow.com/questions/42489918/async-await-inside-arraymap

Reply:
oh right, forgot async function return promise, my bad